src/settings/mod.rs

```diff
@@ -124,6 +123,16 @@ pub struct Settings {
 
 impl Settings {
     pub fn from_configuration(config: Configuration, project_root: &Path) -> Result<Self> {
+        if let Some(required_version) = &config.required_version {
+            if &**required_version != CARGO_PKG_VERSION {
+                return Err(anyhow!(
+                    "Required version `{}` does not match the running version `{}`",
+                    &**required_version,
+                    CARGO_PKG_VERSION
+                ));
+            }
+        }
```

Comment:
Is the idea here that we now validate when reading the config, rather than when transforming it into `Settings`? I wonder why I didn't do it this way initially. I'm getting Chesterton's fence vibes.

Reply:
I am sorry I cannot tell you why you wrote it this way ... looking at it just doesn't make sense to me ... as I explained in the commit message in Rust you ideally manage to make illegal states unrepresentable ... since `Settings` is used all over the codebase it only makes sense that `Settings` denotes valid settings as opposed to settings that may be invalid.

Reply:
Are you saying you don't like my code...? (Joking)

Reply:
Thank you :) I was only kidding but maybe I'll just say a few quick things:

- I try to stay unattached to my code so, of course, always feel free to suggest improvements.
- To me much of building software is tradeoffs and prioritization... Ruff is a big project and so required making a lot of tradeoffs!
- As a result, there's certainly "bad" code in various places, things that are done poorly that I know I'd like to be done differently, things that are done poorly that I've totally forgotten about, etc.
- ...and as such, I'm really appreciative of all the larger refactors you've taken on! They've made the project much better. Some of them aligned with things I wanted to change, some of them hadn't even occurred to me.

Reply:
I know you were kidding^^

Right ... I just often don't like making tradeoffs and try to avoid them and end up going down some rabbit hole.